src/main/java/io/jenkins/plugins/credentials/secretsmanager/factory/CredentialsFactory.java

```diff
@@ -0,0 +1,151 @@
+package io.jenkins.plugins.credentials.secretsmanager.factory;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.amazonaws.services.secretsmanager.model.GetSecretValueRequest;
+import com.amazonaws.services.secretsmanager.model.GetSecretValueResult;
+import com.cloudbees.plugins.credentials.CredentialsUnavailableException;
+import com.cloudbees.plugins.credentials.SecretBytes;
+import com.cloudbees.plugins.credentials.common.StandardCredentials;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.util.Secret;
+import io.jenkins.plugins.credentials.secretsmanager.Messages;
+import io.jenkins.plugins.credentials.secretsmanager.factory.certificate.AwsCertificateCredentials;
+import io.jenkins.plugins.credentials.secretsmanager.factory.ssh_user_private_key.AwsSshUserPrivateKey;
+import io.jenkins.plugins.credentials.secretsmanager.factory.string.AwsStringCredentials;
+import io.jenkins.plugins.credentials.secretsmanager.factory.username_password.AwsUsernamePasswordCredentials;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.logging.Logger;
+
+public abstract class CredentialsFactory {
+
+    private CredentialsFactory() {
+
+    }
+
+    /**
+     * Construct a Jenkins credential from a Secrets Manager secret.
+     *
+     * @param name the secret's name (must be unique)
```

Comment:
Nit: might be useful to be specific - I believe it needs to be unique within an AWS account?

Reply:
Updated the doc